i3lock.c

```diff
@@ -210,6 +212,16 @@ static void clear_pam_wrong(EV_P_ ev_timer *w, int revents) {
 
     /* Now free this timeout. */
     STOP_TIMER(clear_pam_wrong_timeout);
+
+    /* retry with input done during pam verification */
+    if (retry_verification) {
+        retry_verification = false;
+
+        password[input_position] = '\0';
+        unlock_state = STATE_KEY_PRESSED;
+        redraw_screen();
+        input_done();
```

Comment:
Could you refactor the 4 lines above into a function so that we can avoid the duplication please?